pixeltable/index/embedding_index.py

```diff
@@ -119,15 +127,28 @@ def records_value_errors(self) -> bool:
 
     def get_index_sa_type(self, val_col_type: ts.ColumnType) -> sql.types.TypeEngine:
         assert isinstance(val_col_type, ts.ArrayType) and val_col_type.shape is not None
-        vector_size = val_col_type.shape[0]
-        assert vector_size is not None
-        return pgvector.sqlalchemy.Vector(vector_size)
+        assert len(val_col_type.shape) == 1
+        vector_length = val_col_type.shape[0]
+        assert vector_length is not None
+
+        if vector_length <= MAX_EMBEDDING_VECTOR_LENGTH:
```

Comment:
As I commented on slack - I think there's a strong argument for always defaulting to HalfVector for new embeddings - and making it a property of the embedding index that the user can control. Let's discuss at standup.